packages/backend/server/schema.prisma

```diff
@@ -618,3 +620,43 @@ model Blob {
   @@id([workspaceId, key])
   @@map("blobs")
 }
+
+enum NotificationType {
+  Mention
+  Invitation
+  InvitationAccepted
+  InvitationBlocked
+}
+
+enum NotificationLevel {
+  // Makes a sound and appears as a heads-up notification
+  High
+  // Makes a sound
+  Default
+  // Makes no sound
+  Low
+  Min
+  None
+}
+
+model Notification {
+  id            String            @id @default(uuid()) @db.VarChar
+  userId        String            @map("user_id") @db.VarChar
+  createdAt     DateTime          @default(now()) @map("created_at") @db.Timestamptz(3)
+  expiredAt     DateTime?         @map("expired_at") @db.Timestamptz(3)
```

Comment:
I do recall a bad experience on such practice

we will do outdated notifications cleanup regarding `expiredAt`

```ts
DELETE FROM TABLE notifications where `expired_at` is not null and `expired_at` < now() - '180d'
```

it will seq scan the full table, and of course the table will contains millions of records because it's notifications. so the modification operation will cause the whole table been locked for a very long time.

Reply:
but index on `[expiredAt, read]` won't make it better because expiredAt is nullable, and read is boolean, them won't make the index works appropriate.

Reply:
index on `[createdAt, read]` will be better because `createdAt` will be linear values, which is the best candidate of indexes